src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputParameterConverter.cs

```diff
@@ -123,7 +123,8 @@ public static InputParameter CreateInputParameter(ref Utf8JsonReader reader, str
 
         private static SerializationFormat GetSerializationFormat(InputType parameterType, RequestLocation requestLocation)
         {
-            if (parameterType is InputPrimitiveType { Kind: InputTypeKind.DateTime })
+            var affectType = parameterType is InputListType listType? listType.ElementType : parameterType is InputDictionaryType dictionaryType? dictionaryType.ValueType: parameterType;
```

Comment:
This line of code now is quite cumbersome because it has nested ternary operators.
Could we do something like switch statements?
```csharp
var affectType = parameterType switch
{
InputListType listType => listType.ElementType,
InputDictionaryType dictionaryType => dictionaryType.ValueType,
_ => parameterType
};
```

Reply:
Good suggestion. updated. Thanks